src/main/groovy/io/github/cosmicsilence/scalafix/ScalafixPlugin.groovy

```diff
@@ -90,31 +87,43 @@ class ScalafixPlugin implements Plugin<Project> {
                 prop.split('\\s*,\\s*').findAll { !it.isEmpty() }.toList()
             }))
             scalafixTask.mode = taskMode
-            scalafixTask.scalaVersion = sourceSet.scalaVersion.get()
-            scalafixTask.classpath = sourceSet.fullClasspath.collect { it.path }
-            scalafixTask.compileOptions = sourceSet.compilerOptions
-            scalafixTask.semanticdbConfigured = extension.isSemanticdbEnabled()
+            scalafixTask.sourceSetName = sourceSet.name
+            scalafixTask.scalaVersion.set(project.provider({ resolveScalaVersion(sourceSet, extension) }))
+            scalafixTask.classpath.set(project.provider( { sourceSet.fullClasspath.collect { it.path } }))
+            scalafixTask.compileOptions.set(project.provider({ sourceSet.compilerOptions }))
+            scalafixTask.semanticdbConfigured = extension.semanticdbEnabled
 
-            if (extension.isSemanticdbEnabled()) {
+            if (extension.semanticdbEnabled) {
                 scalafixTask.dependsOn sourceSet.compileTask
             }
         })
 
         mainTask.dependsOn taskProvider
     }
 
-    private void configureSemanticdbCompilerPlugin(Project project, ScalaSourceSet sourceSet) {
-        def semanticDbCoordinates = ScalafixProps.getSemanticDbArtifactCoordinates(sourceSet.scalaVersion.get(),
-                Optional.ofNullable(project.scalafix.semanticdb.version.getOrNull()))
+    private void configureSemanticdbCompilerPlugin(Project project, ScalaSourceSet sourceSet, ScalafixExtension extension) {
+        def scalaVersion = resolveScalaVersion(sourceSet, extension)
```

Comment:
This one cannot be avoided (unless there is a Gradle trick that I'm not aware of) as we need to resolve the semanticdb-scalac compiler plugin dependency and pass the path to the jar as a compiler option to the `ScalaCompile` task. And that seems to only be possible during the configuration phase. 

In order to allow users to workaround issues/conflicts caused by the early dependency resolution, I added the `scalaVersion` property to the scalafix extension as per [suggestion](https://github.com/cosmicsilence/gradle-scalafix/issues/49#issue-915047973) from @borissmidt. If that property is set, the Scala version detection step is skipped

Reply:
Turns out, it was possible to defer the resolution of the `semanticdb-scalac` compiler plugin dependency to the execution phase 🤦 . I tested that with the `scoverage` (see #19) as well as `gradle-crossbuild-scala` (see #49) and no conflicts were found. As a result, I won't be adding the `scalaVersion` property to the scalafix extension anymore. It's more convenient and safer for users to rely on the Scalafix plugin to figure out the Scala version. I may reconsider this if other conflicts are identified.